anndata/_core/sparse_dataset.py

```diff
@@ -131,13 +147,21 @@ def _get_intXslice(self, row: int, col: slice) -> ss.csr_matrix:
         )[:, col]
 
     def _get_sliceXslice(self, row: slice, col: slice) -> ss.csr_matrix:
+        row = _normalize_slice(row, self.shape[0])
+        col = _normalize_slice(col, self.shape[1])
+
         out_shape = (
             slice_len(row, self.shape[0]),
             slice_len(col, self.shape[1]),
         )
+
         if out_shape[0] == 1:
             return self._get_intXslice(slice_as_int(row, self.shape[0]), col)
         elif out_shape[1] == self.shape[1] and out_shape[0] < self.shape[0]:
+            if row.step == 1:
+                return ss.csr_matrix(
+                    self._get_contiguous_compressed_slice(row), shape=out_shape
+                )
             return self._get_arrayXslice(np.arange(*row.indices(self.shape[0])), col)
```

Comment:
if it is -1, why can't we do this process in reverse?

Reply:
It's a little more complicated to do this in reverse, since it's not just a contiguous slice of the underlying arrays (it could be in reverse, but then it's not sorted). I'm going to give this a shot, but may end up only optimizing forward slices in this PR.

Reply:
I'm going to leave this for now, just for expediency. Please do make a PR on this!